src/os/bluestore/bluestore_tool.cc

```diff
@@ -219,6 +227,60 @@ void inferring_bluefs_devices(vector<string>& devs, std::string& path)
   }
 }
 
+void bluefs_import(
+  string input_file,
+  string dest_file,
+  CephContext *cct,
+  const string& path,
+  const vector<string>& devs)
+{
+  int r;
+  std::ifstream f(input_file.c_str(), std::ifstream::binary);
+  if (!f) {
+    r = -errno;
+    cerr << "open " << input_file.c_str() << " failed: " << cpp_strerror(r) << std::endl;
+    exit(EXIT_FAILURE);
+  }
+
+  std::unique_ptr<BlueFS> bs = std::unique_ptr<BlueFS>(open_bluefs(cct, path, devs));
+
+  BlueFS::FileWriter *h;
+  fs::path file_path(dest_file);
+  const string dir = file_path.parent_path();
+  const string file_name = file_path.filename();
+  r = bs->open_for_write(dir, file_name, &h, true);
+  if (r < 0) {
+    cerr << "Open " << dir << "/" << file_name << " failed: " << cpp_strerror(r) << std::endl;
+    exit(EXIT_FAILURE);
+  }
+  uint64_t dsize = 0;
+  uint64_t fsize = fs::file_size(input_file.c_str());
+  r = bs->stat(dir, file_name, &dsize, NULL);
+  if (r != 0) {
+    cerr << "Get " << dir << "/" << file_name << " stat failed: " << cpp_strerror(r) << std::endl;
+    exit(EXIT_FAILURE);
+  }
+  if (fsize < dsize) {
+     bs->truncate(h, fsize);
+  } 
+  char buf[256];
+  int left = fsize / 256;
+  int size = 256;
+  while (left > -1) {
```

Comment:
max_block = 4096;
while(left) {
size = std::min(max_block, left);

left-=size;
}

looks a bit more elegant implementation

Reply:
Yes, your way is more elegant.
Thank you for your advice